### PR TITLE
feat: add urdu language in ant-design

### PR DESCRIPTION
**Pull request by MuhammadAnas190**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated
- [x] Demo is not needed
- [x] TypeScript definition not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ timeline: true
 
 ---
 
+## 4.16.4
+
+`2021-06-18`
+
+- 🌐 Internationalization
+  - 🇵🇰 Add Urdu locale. [#31035](https://github.com/ant-design/ant-design/pull/31035) [@MuhammadAnas190](https://github.com/MuhammadAnas190)
```
Comment (afc163) [inline CHANGELOG.en-US.md:23]: Don't modify changelog and package version.

File: components/date-picker/locale/ur_PK.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import CalendarLocale from 'rc-picker/lib/locale/en_US';
```
Comment (yoyo837) [inline components/date-picker/locale/ur_PK.tsx:1]: Your need add `rc-picker/lib/locale/ur_PK` first.

File: components/locale/ur_PK.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-template-curly-in-string */
+import Pagination from 'rc-pagination/lib/locale/en_US';
```
Comment (yoyo837) [inline components/locale/ur_PK.tsx:2]: Your need add `rc-pagination/lib/locale/ur_PK` first.

File: CHANGELOG.en-US.md (hunk 1 above)
Comment (MuhammadAnas190) [reply CHANGELOG.en-US.md]: @afc163  reverted

File: components/date-picker/locale/ur_PK.tsx (hunk 2 above)
Comment (MuhammadAnas190) [reply components/date-picker/locale/ur_PK.tsx:1]: @yoyo837  Added please check [https://github.com/react-component/picker/pull/266](https://github.com/react-component/picker/pull/266)

File: components/locale/ur_PK.tsx (hunk 3 above)
Comment (MuhammadAnas190) [reply components/locale/ur_PK.tsx:2]: @yoyo837 added please check [https://github.com/react-component/pagination/pull/342](https://github.com/react-component/pagination/pull/342)

**Comment by yoyo837** [comment]
Waiting for the upstream release new.

File: docs/react/i18n.en-US.md [hunk 4]
```diff
@@ -75,6 +75,7 @@ The following languages are currently supported:
 | Polish                   | pl_PL    |
 | Portuguese (Brazil)      | pt_BR    |
 | Portuguese               | pt_PT    |
+| Urdu                     | ur_PK    |
```
Comment (yoyo837) [inline docs/react/i18n.en-US.md:78]:
```suggestion
| Urdu(Pakistan)                     | ur_PK    |
```


i18n.zh-CN.md also needs this.

**Comment by MuhammadAnas190** [comment]
@yoyo837 @afc163 please let me know when will this PR get merges.

**Comment by yoyo837** [comment]
> @yoyo837 @afc163 please let me know when will this PR get merges.

Still waiting for the upstream release, when it is out, I am happy to remind you.

File: components/date-picker/locale/ur_PK.tsx (hunk 2 above)
Comment (afc163) [reply components/date-picker/locale/ur_PK.tsx:1]: https://github.com/react-component/picker/releases/tag/v2.5.13

File: components/locale/ur_PK.tsx (hunk 3 above)
Comment (afc163) [reply components/locale/ur_PK.tsx:2]: https://github.com/react-component/pagination/releases/tag/v3.1.7

**Comment by yoyo837** [comment]
@MuhammadAnas190 Let's continue this work.

**Comment by MuhammadAnas190** [comment]
Packages are now updated @yoyo837 @afc163

**Comment by yoyo837** [comment]
This may need to switch to `feature` branch.

**Comment by MuhammadAnas190** [comment]
@yoyo837 please can you let me know how to do that.

**Review by afc163 — CHANGES_REQUESTED** [review]
Please send this PR to feature branch.

File: docs/react/i18n.en-US.md [hunk 5]
```diff
@@ -76,6 +76,7 @@ The following languages are currently supported:
 | Polish                   | pl_PL    |
 | Portuguese (Brazil)      | pt_BR    |
 | Portuguese               | pt_PT    |
+| Urdu(Pakistan)           | ur_PK    |
```
Comment (yoyo837) [inline docs/react/i18n.en-US.md:79]: `i18n.zh-CN.md` also need to add this.

`| 乌尔都语(巴基斯坦)           | ur_PK    |`

Comment (MuhammadAnas190) [reply docs/react/i18n.en-US.md:79]: @yoyo837 please check

**Comment by yoyo837** [comment]
Some other commits have been mixed, please rebase and force push.

**Comment by yoyo837** [comment]
Still mixed with other commits.

**Comment by afc163** [comment]
Still mixed with other commits.

**Comment by yoyo837** [comment]
Rebase upstream `feature`, not `master`.

**Comment by MuhammadAnas190** [comment]
My bad, doing that after some chores

**Comment by MuhammadAnas190** [comment]
Rebase *`feature`* is not working as it's highlighting `Current branch master is up to date.`. Let me know should I create a different PR and discard this?

**Comment by MuhammadAnas190** [comment]
Please check now @yoyo837 @afc163

**Comment by afc163** [comment]
图片

Still mixed with other commits.

**Comment by MuhammadAnas190** [comment]
@afc163 @yoyo837 let me know should I create a new PR for the feature as rebasing with master and feature has mixed up the commits.

**Comment by yoyo837** [comment]
> @afc163 @yoyo837 let me know should I create a new PR for the feature as rebasing with master and feature has mixed up the commits.

Yes, a new one checkout from upstream feature.

**Comment by MuhammadAnas190** [comment]
@yoyo837 I already check out twice from feature branch but the issue still persist, instead creating a new PR might solve this issue I guess.

**Comment by MuhammadAnas190** [comment]
@yoyo837 @afc163 created a new PR for this feature https://github.com/ant-design/ant-design/pull/31346

**Pull request closed without merging**